addons/web/controllers/main.py

```diff
@@ -738,6 +738,10 @@ def __init__(self, field_names, row_count=0):
         self.datetime_style = self.workbook.add_format({'text_wrap': True, 'num_format': 'yyyy-mm-dd hh:mm:ss'})
         self.worksheet = self.workbook.add_worksheet()
         self.value = False
+        decimal_separator = get_lang(request.env).decimal_point
+        self.float_format = '0{}00'.format(decimal_separator)
+        decimal_places = [res['decimal_places'] for res in request.env['res.currency'].search_read([], ['decimal_places'])]
+        self.monetary_format = ('0{}' + max(decimal_places) * '0').format(decimal_separator) if decimal_places else self.float_format
```

Comment:
The last line is too complicated, string concatenation, string multiplication, str.format, ternary, *oof*.

```suggestion
        decimal_separator = get_lang(request.env).decimal_point
        self.float_format = f'0{decimal_separator}00'
        decimal_places = [res['decimal_places'] for res in request.env['res.currency'].search_read([], ['decimal_places'])]
        self.monetary_format = f'0{decimal_separator}{max(decimal_places or [2]) * '0'}'
```

tips: starting with py3.9, we can use `max(decimal_places, default=2)` instead, here we have to support 3.7.